### PR TITLE
FastGalleryTemplate. Includes User and Application model.

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
This PR enables a user to easily create a gallery of applications and/ or other resources from a `.yaml` file.

- Provides most of #2313. The rest can be provided when #2315 has been implemented.
- The User and Application Model developed in this PR is also needed for #2314

Todo:

[x] Add User and Application Models
[x] Add FastGalleryTemplate
[x] Add functionality to read configuration of Users and Applications from `.yaml` file
[] Document FastGalleryTemplate

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
You can test the gallery using a command like `panel serve panel/tests/test_fast_gallery_template.py --auto --show`

**Comment by philippjfr** [comment]
Thanks for making a start on this. I would strongly urge you to split up the PR though, especially the config file will need significant discussion and should not be conflated with the gallery and menu.

File: panel/tests/template/fast/test_fast_list_template.py [hunk 1]
```diff
@@ -21,9 +21,8 @@ def test_template_theme_parameter():
 
 def test_app():
     app = FastListTemplate(
-        title="FastListTemplate",
         site="Panel",
-        site_url="https://panel.holoviz.org",
```
Comment (MarcSkovMadsen) [inline panel/tests/template/fast/test_fast_list_template.py:26]: Not related to this PR. But improves the manual test.

File: panel/tests/test_template.py [hunk 2]
```diff
@@ -18,16 +19,16 @@
 from panel.layout import Row
 from panel.pane import HoloViews, Markdown
 from panel.template import (
-    BootstrapTemplate, GoldenTemplate, MaterialTemplate, ReactTemplate,
+    BootstrapTemplate, FastListTemplate, FastGridTemplate, GoldenTemplate, MaterialTemplate, ReactTemplate,
```
Comment (MarcSkovMadsen) [inline panel/tests/test_template.py:22]: Not relevant to this PR. But I discovered these templates where not tested in same way as the other templates.

**Comment by philippjfr** [comment]
As discussed I'll close but would love to see this in a separate repo.

**Pull request closed without merging**